src/main/java/com/exasol/projectkeeper/validators/changelog/dependencies/MavenDependencyComparator.java

```diff
@@ -0,0 +1,66 @@
+package com.exasol.projectkeeper.validators.changelog.dependencies;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.maven.model.Dependency;
+
+/**
+ * This class compares two lists of maven dependencies.
+ */
+public class MavenDependencyComparator {
+
+    /**
+     * Compare two lists of maven dependencies.
+     * 
+     * @param oldDependencies old list of dependencies
+     * @param newDependencies new list of dependencies
+     * @return list of dependency changes
+     */
+    public List<DependencyChange> compare(final List<Dependency> oldDependencies,
+            final List<Dependency> newDependencies) {
+        final List<DependencyChange> report = new ArrayList<>();
+        final Map<String, Dependency> oldDependenciesMap = createMapWithDependencyKey(oldDependencies);
+        final Map<String, Dependency> newDependenciesMap = createMapWithDependencyKey(newDependencies);
+        oldDependenciesMap.forEach((key, oldDependency) -> {
+            if (newDependenciesMap.containsKey(key)) {
+                addDependencyUpdateIfVersionChanged(report, newDependenciesMap.get(key), oldDependency);
+            } else {
+                addDependencyRemove(report, oldDependency);
+            }
+        });
+        newDependenciesMap.forEach((key, newDependency) -> {
+            if (!oldDependenciesMap.containsKey(key)) {
+                addDependencyAdd(report, newDependency);
+            }
+        });
+        return report;
+    }
+
+    private void addDependencyAdd(final List<DependencyChange> report, final Dependency newDependency) {
```

Comment:
`addDependencyAdd` sounds funny. Maybe the classes can be renamed to "NewDependency", "UpdatedDependendy" and "RemovedDependency"?